### DISCUSSION TITLE
[Question] Send image and text in one message

### DISCUSSION THREAD

**Discussion by hillbicks** [Skylar-Tech/node-red-contrib-matrix-chat#45] [Q&A]
Hey there,

I'm struggling to send a message with text and a picture to a matrix chat. I tried to send the mail as HTML formatted, but apparently there is an open [issue ](https://github.com/vector-im/element-android/issues/1531)that this doesn't work at the moment. Well, it kinda works, but only in the android client it seems. Both element desktop and web clients are not showing the images. 

Is there a way to send (formatted) text alongside a picture with this addon for node-red?

PS: I hope it's ok I opened up an issue for a question, but since there is no discussion tab available, I figured I ask here.

**Comment by skylord123** [comment]
You can't attach a message to the image, you will need to use a send message node either before or after the image sends. I have this attached to my security system and what I do is send a message of the event (example: Front door opened at 6:00AM) then I send the nearest security camera image to the room.

You can set `msg.filename` or `msg.body` on the input of the send image node. In practice the `body` gets used for both the filename and as text that renders in the client but only if the image is replied to:

If you reply to an image with a body set it will display the body next to the image when replying like so (in this case `msg.body` was set to "this is the test body"):
![image](https://user-images.githubusercontent.com/3412313/153231653-067f27a9-1685-4737-aa7c-aec4c09c3e1a.png)
And once it is posted it looks like this:
![image](https://user-images.githubusercontent.com/3412313/153231788-e3c0c0eb-7bc3-4e85-bbcc-e9ced1e30ddd.png)

But as you can see the body didn't render on the initial image and only on replies. In this case I also set the filename to `filename.png` but if you download the image from element it auto fills the name as "this is the body.png". The filename is only used for naming the file in the Matrix storage (and is pretty much useless because of this but I left it anyways).

According to the [Matrix Spec](https://spec.matrix.org/v1.1/client-server-api/#fallback-for-mimage-mvideo-maudio-and-mfile) the body is:

> A textual representation of the image. This could be the alt text of the image, the filename of the image, or some kind of content description for accessibility e.g. ‘image attachment’.

Hope this answers your question :)

**Reply by hillbicks** [reply]
Well, I guess the tldr is: Nope, not possible at the moment. 

One use case I have is similar to yours. Once a person was detected, send a message via matrix and attach the screenshot from camera. It never occured to me that this might be a limitation of matrix itself. 

I just found this open [issue](https://github.com/matrix-org/matrix-doc/issues/906) from 2017!!! which links to two open MSCs 
https://github.com/matrix-org/matrix-doc/pull/2529
https://github.com/matrix-org/matrix-doc/pull/2530

A real bummer, that such a seemingly simple task doesn't seem to be possible at the moment. I guess I have to watch for this issue to be solved, or the html one to be fixed (which I would prefer). 

Come to think of it: What about markdown? Maybe that would be a way to solve this. Is markdown supported? Couldn't find any info on that. 

I just tested a bit with just sending an image and this is driving me crazy 😂 

The layout is different, depending on the client I use to send image with AND the chatclient. Well, at least when sending the images via node red, matrix-commander seems to be consistent here. 

I made a couple of screenshots to illustrate what I'm talking about. 

https://imgur.com/a/DDruzwE

Have you noticed this as well?

All I want to achieve is to send a message with corresponding text, that at least looks like it belongs together.

Ok, enough with my ranting......

**Comment by skylord123** [comment]
Went ahead and setup discussions and migrated this issue. Thanks for bringing that up, completely slipped my mind.
